src/v/kafka/server/handlers/produce.cc

```diff
@@ -34,6 +34,11 @@
 #include <boost/container_hash/extensions.hpp>
 #include <fmt/ostream.h>
 
+#include <chrono>
+#include <cstdint>
+#include <memory>
+#include <ranges>
+
```

Comment:
nitpick: commit says we're removing unused header includes, but the diff appears to be adding four new ones?

Reply:
Not really a nitpick, a good observation! 

This is an older change and Michal submitted a change which removed the same headers so this change is pointless after I rebased it (don't ask my why it adds those headers). I will delete it.